packages/flutter/lib/src/widgets/platform_menu_bar.dart

```diff
@@ -0,0 +1,886 @@
+  /// of children of the [PlatformMenu] it belongs to, and `count` is the number
+  /// of children in the [PlatformMenu] it belongs to.
+  Iterable<Map<String, Object?>> toChannelRepresentation(
+    DefaultPlatformMenuDelegate delegate, {
+    required int index,
+    required int count,
+    required DefaultPlatformMenuDelegateSerializableIdGenerator getId,
+  });
+}
+
+/// The platform menu delegate that handles the built-in macOS platform menu
+/// generation using the 'flutter/menu' channel.
+///
+/// An instance of this class is set on [WidgetsBinding.platformMenuDelegate] by
+/// default when the [WidgetsBinding] is initialized.
+///
+/// See also:
+///
+///  * [PlatformMenuBar], the widget that adds a platform menu bar to an
+///    application.
+///  * [PlatformMenu], the class that describes a menu item with children
+///    that appear in a cascading menu.
+///  * [PlatformMenuItem], the class that describes the leaves of a menu
+///    hierarchy.
+class DefaultPlatformMenuDelegate extends PlatformMenuDelegate {
+  /// Creates a const [DefaultPlatformMenuDelegate].
+  ///
+  /// The optional [channel] argument defines the channel used to communicate
+  /// with the platform. It defaults to [SystemChannels.menu] if not supplied.
+  DefaultPlatformMenuDelegate({MethodChannel? channel})
+      : channel = channel ?? SystemChannels.menu,
+        _idMap = <int, MenuItem>{} {
+    this.channel.setMethodCallHandler(_methodCallHandler);
+  }
+
+  // Map of distributed IDs to menu items.
+  final Map<int, MenuItem> _idMap;
+  // An ever increasing value used to dole out IDs.
+  int _serial = 0;
+  // The context used to "lock" this delegate to a specific instance of
+  // PlatformMenuBar to make sure there is only one.
+  BuildContext? _lockedContext;
+
+  @override
+  void clearMenus() => setMenus(<MenuItem>[]);
+
+  @override
+  void setMenus(List<MenuItem> topLevelMenus) {
+    _idMap.clear();
+    final List<Map<String, Object?>> representation = <Map<String, Object?>>[];
+    if (topLevelMenus.isNotEmpty) {
+      int index = 0;
+      for (final MenuItem childItem in topLevelMenus) {
+        if (childItem is! DefaultPlatformMenuDelegateSerializable) {
+          throw UnimplementedError(
+              'Tried to serialize a menu item that was not a DefaultPlatformMenuDelegateSerializer');
+        }
+        representation.addAll((childItem as DefaultPlatformMenuDelegateSerializable)
+            .toChannelRepresentation(this, index: index, count: topLevelMenus.length, getId: _getId));
+        index += 1;
+      }
+    }
+    channel.invokeMethod<void>(_kMenuSetMethod, representation);
+  }
+
+  /// Defines the channel that the [DefaultPlatformMenuDelegate] uses to
+  /// communicate with the platform.
+  ///
+  /// Defaults to [SystemChannels.menu].
+  final MethodChannel channel;
+
+  /// Get the next serialization ID.
+  ///
+  /// This is called by each DefaultPlatformMenuDelegateSerializer when
```

Comment:
DefaultPlatformMenuDelegateSerializer is public, indicating that a custom MenuItem in a file outside of this class (e.g. in an app) may be allowed to implement it. However, they wouldn't be able to call this method since it is private. Should DefaultPlatformMenuDelegateSerializer be private as well?

Reply:
Ah, I see. This method is actually teared off and passed to the DefaultPlatformMenuDelegateSerializer. Never mind my comment above then.